### PR TITLE
BlockHandler initialisation is a constant expression

### PR DESCRIPTION
**Pull request by tigerw**
If we can't make it all namespaces, this is the next best I guess.

+ Tag handlers constexpr, const as needed
+ Inherit constructors
* Privatise handler functions

### CODE REVIEW

**Comment by tigerw** [comment]
```
/root/cuberite/src/Blocks/BlockFence.h:28:15: error: invalid case style for private member 'MAX_COORD' [readability-identifier-naming,-warnings-as-errors]
        const double MAX_COORD = 0.6;
                     ^~~~~~~~~
                     m_MAX_COORD
```

File: src/Blocks/BlockHandler.h [hunk 1]
```diff
@@ -25,9 +25,11 @@ class BlockTypeRegistry;
 class cBlockHandler
 {
 public:
-	cBlockHandler(BLOCKTYPE a_BlockType);
 
-	virtual ~cBlockHandler() {}
```
Comment (peterbell10) [inline src/Blocks/BlockHandler.h:30]: This should still have a virtual destructor, even if you don't think you'll need it.

File: src/Blocks/BlockFence.h [hunk 2]
```diff
@@ -17,22 +17,17 @@ class cBlockFenceHandler:
...
 
+	using Super::Super;
+
+private:
+
 	// These are the min and max coordinates (X and Z) for a straight fence.
 	// 0.4 and 0.6 are really just guesses, but they seem pretty good.
 	// (0.4 to 0.6 is a fence that's 0.2 wide, down the center of the block)
 	const double MIN_COORD = 0.4;
 	const double MAX_COORD = 0.6;
```
Comment (peterbell10) [inline src/Blocks/BlockFence.h:28]: Making it `static constexpr` should silence the warnings IIRC.

**Review by peterbell10 — CHANGES_REQUESTED** [review]

File: src/Blocks/BlockFence.h (hunk 2 above)
Comment (tigerw) [reply src/Blocks/BlockFence.h:28]: Thanks!

File: src/Blocks/BlockHandler.h (hunk 1 above)
Comment (tigerw) [reply src/Blocks/BlockHandler.h:30]: aww but I looked forward to deleting this. I'm not doing any dynamic allocation, so why do I need one?

Comment (peterbell10) [reply src/Blocks/BlockHandler.h:30]: AFAIK there's no reason not to include it, and some potentially nasty bugs avoided by including it.

Comment (tigerw) [reply src/Blocks/BlockHandler.h:30]: Hm, it doesn't let me add a destructor to constexpr

Comment (peterbell10) [reply src/Blocks/BlockHandler.h:30]: Hrm, I guess it's not possible for `constexpr` then. Oh well.

**Review by bearbin — COMMENTED** [review]
Soft-approve: the changes look good in principle but I haven't read through every line.

**Review by peterbell10 — APPROVED** [review]

**Pull request merged**